test/liblll/Compiler.cpp

```diff
@@ -121,6 +122,504 @@ BOOST_AUTO_TEST_CASE(switch_inconsistent_return_count)
+    "00",
+    "01",
+    "02",
+    "03",
+    "04",
+    "05",
+    "06",
+    "07",
+    "08",
+    "09",
+    "0a",
+    "0b",
+    "10",
+    "11",
+    "12",
+    "13",
+    "14",
+    "15",
+    "16",
+    "17",
+    "18",
+    "19",
+    "1a",
+    "20",
+    "30",
+    "31",
+    "32",
+    "33",
+    "34",
+    "35",
+    "36",
+    "37",
+    "38",
+    "39",
+    "3a",
+    "3b",
+    "3c",
+    "3d",
+    "3e",
+    "40",
+    "41",
+    "42",
+    "43",
+    "44",
+    "45",
+    "50",
+    "51",
+    "52",
+    "53",
+    "54",
+    "55",
+    "56",
+    "57",
+    "58",
+    "59",
+    "5a",
+    "5b",
+    "60",
+    "61",
+    "62",
+    "63",
+    "64",
+    "65",
+    "66",
+    "67",
+    "68",
+    "69",
+    "6a",
+    "6b",
+    "6c",
+    "6d",
+    "6e",
+    "6f",
+    "70",
+    "71",
+    "72",
+    "73",
+    "74",
+    "75",
+    "76",
+    "77",
+    "78",
+    "79",
+    "7a",
+    "7b",
+    "7c",
+    "7d",
+    "7e",
+    "7f",
+    "80",
+    "81",
+    "82",
+    "83",
+    "84",
+    "85",
+    "86",
+    "87",
+    "88",
+    "89",
+    "8a",
+    "8b",
+    "8c",
+    "8d",
+    "8e",
+    "8f",
+    "90",
+    "91",
+    "92",
+    "93",
+    "94",
+    "95",
+    "96",
+    "97",
+    "98",
+    "99",
+    "9a",
+    "9b",
+    "9c",
+    "9d",
+    "9e",
+    "9f",
+    "a0",
+    "a1",
+    "a2",
+    "a3",
+    "a4",
+    "f0",
+    "f1",
+    "f2",
+    "f3",
+    "f4",
+    "fa",
+    "fd",
+    "fe",
+    "ff"
+  };
+
+  std::vector<std::string> opcodes_lll {
+    "{ (asm STOP) }",
+    "{ (asm ADD) }",
+    "{ (asm MUL) }",
+    "{ (asm SUB) }",
+    "{ (asm DIV) }",
+    "{ (asm SDIV ) }",
+    "{ (asm MOD) }",
+    "{ (asm SMOD) }",
+    "{ (asm ADDMOD) }",
+    "{ (asm MULMOD) }",
+    "{ (asm EXP) }",
+    "{ (asm SIGNEXTEND) }",
+    "{ (asm LT) }",
+    "{ (asm GT) }",
+    "{ (asm SLT) }",
+    "{ (asm SGT) }",
+    "{ (asm EQ) }",
+    "{ (asm ISZERO) }",
+    "{ (asm AND) }",
+    "{ (asm OR) }",
+    "{ (asm XOR) }",
+    "{ (asm NOT) }",
+    "{ (asm BYTE) }",
+    "{ (asm KECCAK256) }",
+    "{ (asm ADDRESS) }",
+    "{ (asm BALANCE) }",
+    "{ (asm ORIGIN) }",
+    "{ (asm CALLER) }",
+    "{ (asm CALLVALUE) }",
+    "{ (asm CALLDATALOAD) }",
+    "{ (asm CALLDATASIZE) }",
+    "{ (asm CALLDATACOPY) }",
+    "{ (asm CODESIZE) }",
+    "{ (asm CODECOPY) }",
+    "{ (asm GASPRICE) }",
+    "{ (asm EXTCODESIZE)}",
+    "{ (asm EXTCODECOPY) }",
+    "{ (asm RETURNDATASIZE) }",
+    "{ (asm RETURNDATACOPY) }",
+    "{ (asm BLOCKHASH) }",
+    "{ (asm COINBASE) }",
+    "{ (asm TIMESTAMP) }",
+    "{ (asm NUMBER) }",
+    "{ (asm DIFFICULTY) }",
+    "{ (asm GASLIMIT) }",
+    "{ (asm POP) }",
+    "{ (asm MLOAD) }",
+    "{ (asm MSTORE) }",
+    "{ (asm MSTORE8) }",
+    "{ (asm SLOAD) }",
+    "{ (asm SSTORE) }",
+    "{ (asm JUMP ) }",
+    "{ (asm JUMPI ) }",
+    "{ (asm PC) }",
+    "{ (asm MSIZE) }",
+    "{ (asm GAS) }",
+    "{ (asm JUMPDEST) }",
+    "{ (asm PUSH1) }",
```

Comment:
Hm, these actually seem to show another bug. They do insert the correct opcode, but there is no possible way to push an immediate, because having a number there will include the PUSH opcode too:
- `(asm PUSH1)` -> 0x60
- `(asm PUSH1 0x42)` -> 0x60 0x60 0x42
- `(asm 0x42)` -> 0x60 0x42

Reply:
I'll push a patch which removes that from assembly, but these tests should be changed to not include `PUSHnn`, but rather have a literal each for all widths (0x42, 0x4242, ..)

Reply:
Fixed in #3070.